company.el

```diff
@@ -2268,11 +2280,12 @@ character, stripping the modifiers.  That character must be a digit."
 (defvar-local company-last-metadata nil)
 
 (defun company-fetch-metadata ()
-  (let ((selected (nth company-selection company-candidates)))
-    (unless (eq selected (car company-last-metadata))
-      (setq company-last-metadata
-            (cons selected (company-call-backend 'meta selected))))
-    (cdr company-last-metadata)))
+  (when company-selection
+    (let ((selected (nth company-selection company-candidates)))
```

Comment:
I think that when nothing is selected, it makes sense to show the metadata for the first candidate anyway.
This way you'll see useful info right away and I don't see any downsides to it.
Therefore I suggest deleting `(when company-selection` and instead doing

    (let ((selected (nth (or company-selection 0) company-candidates)))

Reply:
I kinda worry that it might be confusing.

But if it's really useful in your experience, let's keep it that way for now.